File: skills/writing-tests-that-can-fail/evals/fixtures/d-account-py/fixture_d_account.py
"""Account lifecycle service."""

from dataclasses import dataclass


@dataclass
class Account:
    user_id: str
    email: str
    status: str  # "active" | "inactive"


class AccountRepo:
    def get(self, user_id: str) -> Account:
        raise NotImplementedError("real DB access")

    def set_status(self, user_id: str, status: str) -> None:
        raise NotImplementedError("real DB access")


class Mailer:
    def send_goodbye(self, email: str) -> None:
        raise NotImplementedError("real email send")


class AccountService:
    def __init__(self, repo: AccountRepo, mailer: Mailer) -> None:
        self._repo = repo
        self._mailer = mailer

    def deactivate(self, user_id: str) -> None:
        account = self._repo.get(user_id)
        if account.status == "inactive":
            return
        self._repo.set_status(user_id, "inactive")
        self._mailer.send_goodbye(account.email)

File: evals/writing-tests-that-can-fail/fixtures/d-account-js/fixture_d_account.mjs
// Account lifecycle service.

export class AccountRepo {
  get(userId) {
    throw new Error("real DB access");
  }

  setStatus(userId, status) {
    throw new Error("real DB access");
  }
}

export class Mailer {
  sendGoodbye(email) {
    throw new Error("real email send");
  }
}

export class AccountService {
  constructor(repo, mailer) {
    this.repo = repo;
    this.mailer = mailer;
  }

  deactivate(userId) {
    const account = this.repo.get(userId);
    if (account.status === "inactive") return;
    this.repo.setStatus(userId, "inactive");
    this.mailer.sendGoodbye(account.email);
  }
}
